sponsorship_compassion/models/contracts_report.py

```diff
@@ -0,0 +1,242 @@
+    def _compute_b2s_letter(self):
+        def get_nb_letter(_partner):
+            return self.env['correspondence'].search_count(
+                [('partner_id', '=', _partner.id),
+                 ('direction', '=', 'Beneficiary To Supporter'),
+                 ('scanned_date', '>', _partner.start_period),
+                 ('scanned_date', '<=', _partner.end_period)])
+
+        for partner in self:
+            nb_letter = get_nb_letter(partner)
+            if partner.is_church:
+                for member in partner.member_ids:
+                    nb_letter += get_nb_letter(member)
+            partner.sr_nb_b2s_letter = nb_letter
+
+    @api.multi
+    def _compute_s2b_letter(self):
+        def get_nb_letter(_partner):
+            return self.env['correspondence'].search_count(
+                [('partner_id', '=', _partner.id),
+                 ('direction', '=', 'Supporter To Beneficiary'),
+                 ('scanned_date', '>', _partner.start_period),
+                 ('scanned_date', '<=', _partner.end_period)])
+
+        for partner in self:
+            nb_letter = get_nb_letter(partner)
+            if partner.is_church:
+                for member in partner.member_ids:
+                    nb_letter += get_nb_letter(member)
+            partner.sr_nb_s2b_letter = nb_letter
+
+    @api.multi
+    def _compute_boy(self):
+        for partner in self:
+            total = self.env['compassion.child'].search_count(
+                [('partner_id', '=', partner.id),
+                 ('gender', '=', 'M')])
+            if partner.is_church:
+                for member in partner.member_ids:
+                    total += self.env['compassion.child'].search_count(
+                        [('partner_id', '=', member.id),
+                         ('gender', '=', 'M')])
+            partner.sr_nb_boy = total
+
+    @api.multi
+    def _compute_girl(self):
+        for partner in self:
+            total = self.env['compassion.child'].search_count(
+                [('partner_id', '=', partner.id),
+                 ('gender', '=', 'F')])
+            if partner.is_church:
+                for member in partner.member_ids:
+                    total += self.env['compassion.child'].search_count(
+                        [('partner_id', '=', member.id),
+                         ('gender', '=', 'F')])
+            partner.sr_nb_girl = total
+
+    @api.multi
+    def _compute_time_scp(self):
+        def get_time_in_scp(_partner):
+            total = 0
+            for contract in self.env['recurring.contract'].search(
+                    [('partner_id', '=', _partner.id),
+                     ('type', 'in', ['S', 'SC'])]):
+                contract_start_date = fields.Date.from_string(
+                    contract.start_date)
+                end_date = fields.Date.from_string(
+                    contract.end_date) if contract.end_date else date.today()
+                delta = end_date - contract_start_date
```

Comment:
Instead of computing the delta like here (and you do the same code at different places), I suggest to add a compute field at the  recurring.contract object level :
```
contract_duration = fields.Integer(compute='_compute_contract_duration', help='Contract duration in days')
```
Then I would count like this:
Contracts that were never active count as 0 day (activation_date is not set)
Contracts that were active counts as `end_date - activation_date` instead of `end_date - start_date`
Contracts that are still active counts as `today - activation_date`